tests/tests_geomstats/test_hyperbolic.py

```diff
@@ -18,478 +19,402 @@
-        """
-        # Riemannian Log then Riemannian Exp
-        # Edge case: two very close points, base_point_2 and point_2,
-        # form an angle < epsilon
-        base_point_intrinsic = gs.array([1.0, 2.0, 3.0])
-        base_point = self.space.from_coordinates(base_point_intrinsic, "intrinsic")
-        point_intrinsic = base_point_intrinsic + 1e-12 * gs.array([-1.0, -2.0, 1.0])
-        point = self.space.from_coordinates(point_intrinsic, "intrinsic")
-
-        log = self.metric.log(point=point, base_point=base_point)
-        result = self.metric.exp(tangent_vec=log, base_point=base_point)
-        expected = point
-
-        self.assertAllClose(result, expected)
-
-    def test_exp_and_log_and_projection_to_tangent_space_general_case(self):
-        """
-        Test that the Riemannian exponential
-        and the Riemannian logarithm are inverse.
-
-        Expect their composition to give the identity function.
-        """
-        # Riemannian Exp then Riemannian Log
-        # General case
-        base_point = gs.array([4.0, 1.0, 3.0, math.sqrt(5)])
-        vector = gs.array([2.0, 1.0, 1.0, 1.0])
-        vector = self.space.to_tangent(vector=vector, base_point=base_point)
-        exp = self.metric.exp(tangent_vec=vector, base_point=base_point)
-        result = self.metric.log(point=exp, base_point=base_point)
-
-        expected = vector
-        self.assertAllClose(result, expected)
-
-    def test_dist(self):
-        # Distance between a point and itself is 0.
-        point_a = gs.array([4.0, 1.0, 3.0, math.sqrt(5)])
-        point_b = point_a
-        result = self.metric.dist(point_a, point_b)
-        expected = 0
-        self.assertAllClose(result, expected)
-
-    def test_exp_and_dist_and_projection_to_tangent_space(self):
-        base_point = gs.array([4.0, 1.0, 3.0, math.sqrt(5)])
-        vector = gs.array([0.001, 0.0, -0.00001, -0.00003])
-        tangent_vec = self.space.to_tangent(vector=vector, base_point=base_point)
-        exp = self.metric.exp(tangent_vec=tangent_vec, base_point=base_point)
-
-        result = self.metric.dist(base_point, exp)
-        sq_norm = self.metric.embedding_metric.squared_norm(tangent_vec)
-        expected = sq_norm
-        self.assertAllClose(result, expected, atol=1e-2)
-
-    def test_geodesic_and_belongs(self):
-        initial_point = gs.array([4.0, 1.0, 3.0, math.sqrt(5)])
-        n_geodesic_points = 100
-        vector = gs.array([1.0, 0.0, 0.0, 0.0])
-
-        initial_tangent_vec = self.space.to_tangent(
-            vector=vector, base_point=initial_point
-        )
-        geodesic = self.metric.geodesic(
-            initial_point=initial_point, initial_tangent_vec=initial_tangent_vec
-        )
-
-        t = gs.linspace(start=0.0, stop=1.0, num=n_geodesic_points)
-        points = geodesic(t)
-        result = gs.all(self.space.belongs(points))
-        self.assertTrue(result)
-
-    def test_geodesic_and_belongs_large_initial_velocity(self):
```

Comment:
also keep this one (larger initial vector: that belongs with a bigger atol)

Reply:
needs data generation that is struck in hypersphere...